### ISSUE TITLE
Can't launch debugger, permission denied on `null/node-cdp-xxxx-xx.sock` due to null `os.tmpdir()`

### ISSUE THREAD

**Issue by timbertson** [microsoft/vscode-js-debug#784]
**Describe the bug**

I'm trying to get started with https://code.visualstudio.com/api/get-started/your-first-extension

When I press f5, the debugger fails to launch with the error in the title:

![Screenshot from 2020-10-04 10-17-58](https://user-images.githubusercontent.com/14172/95003591-75d2a980-062c-11eb-9ff8-19abdf2246a0.png)

Given the `null`, and this code: https://github.com/microsoft/vscode-js-debug/blob/9b24a1420f7ce6b4a587e2386de3646ab4834101/src/targets/node/nodeLauncherBase.ts#L348-L349

I think `os.tempdir()` is returning `null`?

**Workaround**:

I tried running with `env TMPDIR=/tmp code .` and that worked.

**VS Code Version:**

Version: 1.48.2
Commit: a0479759d6e9ea56afa657e454193f72aef85bd0
Date: 2020-08-25T10:56:10.074Z
Electron: 7.3.2
Chrome: 78.0.3904.130
Node.js: 12.8.1
V8: 7.8.279.23-electron.0
OS: Linux x64 5.8.12-200.fc32.x86_64

**Additional context**:

I have node 12.18.3 installed, and it doesn't have the same problem (with TMPDIR unset):

```
$ node -e 'console.log(require("os").tmpdir())'
/tmp
```

So it's unclear whether this is fixed in the newer node version, or whether it's some different in vscode's bundled node.

**Comment by connor4312** [comment]
Huh. I've never run into this before. Thanks for the investigative work here; it seems to be something specific to your setup, perhaps; I've not had a report of this previously.

You didn't mention what distro you're running, but most OS' will apply `~/.profile` globally (i.e. to applications as well). Setting `TMPDIR` in there should resolve the problem for you.

**Comment by timbertson** [comment]
Oh right, Fedora (OS: Linux x64 5.8.12-200.fc32.x86_64 was an overly subtle hint, `fc32` = Fedora Core 32)

I mainly reported it in case it's a bug in the vscode fork of node. Probably hard to tell though, aside from updating the bundled nodejs to >= 12.18.3 and see if this still happens.

**Comment by timbertson** [comment]
I'm using nix, but not nixos. I believe the nix derivation just references official binaries, so it shouldn't be too alien from a regular install

**Comment by connor4312** [comment]
Looking around in VS Code, we use the tmpdir in several other places and do not have null handling in any of them. So I would suggest adding the tmpdir to your `~/.profile`, or in some other way make sure VS Code gets launched with it, to prevent other things from breaking in weird ways.

Fyi @deepak1556 in case you have any suspicion that that is electron-related ^

**Comment by deepak1556** [comment]
Nope I don't see electron intercepting the os modules. Quick check, @timbertson the following will provide an repl based on node vscode embeds, lets see if there is actually a difference.

```
$ ELECTRON_RUN_AS_NODE=1 <path-to>/code.exe -e 'console.log(require("os").tmpdir())'
```

Also can you log `$TMPDIR`, `$TMP`, `$TEMP` environment variables. There is always a fallback for tmpdir in node implementation https://github.com/nodejs/node/blob/v12.14.0/lib/os.js#L133-L136 , so unless one of the above env variables is returning `"null"` the error doesn't seem possible.

**Comment by timbertson** [comment]
Well, that's quite strange:

```
env ELECTRON_RUN_AS_NODE=1 /nix/store/g674qkd3a7ynmqx32d4h6gmm7y8qbl6v-vscode-1.48.2/lib/vscode/code -e 'console.log(require("os").tmpdir())'
/tmp
```

`$TMPDIR`, `$TMP`, and `$TEMP` are all unset (not the empty string, genuinely unset)

Running a `code` and dropping into a JS console (via "Toggle developer tools"), I get the same:

```
> require('os').tmpDir()
<- "/tmp"
```

I found `os.js` in the sources tab, and put a breakpoint on the `return path` line. It triggers when I run the above on the console, but it doesn't trigger when I press `<f5>` (even though I still get the original error message).

So: two strange ideas:
 - os.tempdir or os are somehow referring to a _different_ function in nodeLauncherBase and my console
 - this error is actually coming from some other process, and being funneled back into the main one?

It's small potatoes since there is a workaround, but now it's just so strange that I want to get to the bottom of it...

I don't see nodeLauncherBase in my sources tab, is there a way to set breakpoints in it? I assume it all ends up in the minified .js I see in the stacktrace.

**Comment by timbertson** [comment]
Aha, figured it out.

> unless one of the above env variables is returning "null" the error doesn't seem possible.

Indeed... :thinking: 

Debugging wasn't getting me very far so I took a whole copy of the installed `code` directory, bundled extensions and all. I modified the minified `extension.js` to log `process.env` in the code that determines the socket path. Surprisingly, there's a whole lot of actual `"null"` strings:

![Screenshot from 2020-10-11 11-46-08](https://user-images.githubusercontent.com/14172/95667846-5ea93400-0bb7-11eb-8723-8cd17d215c1a.png)

Logging `process.env` at extension _load_ time instead of during this code, the nulls aren't yet there. So it looks like someone's messing with `process.env` between extension load and when this code actually runs. I tried disabling extensions (sorry, should have done this first) and it turns out the problem goes away.

The most obvious candidate from the small number of extensions I use was clearly [vscode-direnv](https://github.com/rubymaniac/vscode-direnv), and indeed disabling just that extension fixes it. So I'll go file a bug over there :)
